src/main/java/io/vertx/core/impl/DeploymentManager.java

```diff
@@ -79,13 +99,47 @@ public void deployVerticle(Verticle verticle, DeploymentOptions options,
       throw new IllegalArgumentException("Can't specify isolatedClasses for already created verticle");
     }
     ContextImpl currentContext = vertx.getOrCreateContext();
-    doDeploy("java:" + verticle.getClass().getName(), generateDeploymentID(), options, currentContext, currentContext, completionHandler,
-        getCurrentClassLoader(), verticle);
+    ClassLoader cl = getClassLoader(options, currentContext);
+    currentContext.<Set<Verticle>>executeBlocking(fut -> {
```

Comment:
do we really need `executeBlocking` here ?

if we do, we should probably say that the thread calling `Supplier#get` is not the same than the thread that will execute the verticle, specially with an event loop based verticle.